src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -86,6 +87,21 @@ internal static void ExtractSegmentIdentifierAndParenthesisExpression(string seg
                 identifier = segmentText;
                 parenthesisExpression = null;
             }
+            else if (segmentText[segmentText.Length - 1] == ':')
+            {
+                if (segmentText.Length < 2 ||
+                    segmentText[segmentText.Length - 2] != ')')

```

Comment:
What about open paren in a key value?

Reply:
Not changing at this point to mitigate risk. The approach I was considering was to pass the entire segment text as identifier if the parentheses didn't match (here if text did not end in parenthesis), however, I was not confident on how TryCreateSegment functions would handle such input.
To reduce scope, I am keeping this out of the PR and creating separate issue for that. 

---
In reply to: [367191420](https://github.com/OData/odata.net/pull/1621#discussion_r367191420) [](ancestors = 367191420)